File: s2/2312/2312.py
import sys
n = int(sys.stdin.readline())
todo = []
for i in range(n):
    todo.append(int(sys.stdin.readline()))
check_prime = [True for _ in range(max(todo)+1)]
prime_num = []
for i in range(2, max(todo) + 1):
    if check_prime[i]:
        prime_num.append(i)
        temp = i*2
        while temp <= max(todo):
            check_prime[temp] = False
            temp += i
for i in range(n):
    k = todo[i]
    for j in prime_num:
        count = 0
        if k == 1:
            break
        while k%j == 0:
            count += 1
            k //= j
        if count != 0:
            print(j, count)

File: s2/2312/2312.cpp
#include<cstdio>
#include<vector>
using namespace std;

int main()
{
	int n;
	scanf("%d", &n);
	int *todo = new int [n];
	int max = 0;
	for(int i=0; i<n; i++)
	{
		scanf("%d", &todo[i]);
		max = todo[i] > max ? todo[i] : max;
	}
	bool *check_prime = new bool [max+1];
	for(int i=0; i<=max; i++)
		check_prime[i] = true;
	vector<int> prime_number;
	for(int i=2; i<=max; i++)
	{
		if(check_prime[i])
		{
			prime_number.push_back(i);
			int temp = i*2;
			while (temp <=max)
			{
				check_prime[temp] = false;
				temp += i;
			}
		}
	}
	vector<int>::iterator iter;
	for(int i=0; i<n; i++)
	{
		int k = todo[i];
		for(iter = prime_number.begin(); iter != prime_number.end(); iter++)
		{
			int prime = *iter;
			if(k==1)
				break;
			int count = 0;
			while (k%prime == 0)
			{
				count += 1;
				k /= prime;
			}
			if(count!=0)
				printf("%d %d\n", prime, count); 
		}
	}

	delete[] todo;
	delete[] check_prime;
	return 0;
}
